### PR TITLE
Feature/sql upsert improvements

### PR DESCRIPTION
**Pull request by JeppeBlixen**
Various improvements to sql upsert

- Refactor to use sql temp tables
- Refactor to use contextmanager, to use 'with connect as conn' syntax
- Add join col filter
- Add overwrite if target is empty
- Add documetnaion of upsert

### CODE REVIEW

File: tests/cluster/sql/test_deliverysql.py [hunk 1]
```diff
@@ -213,13 +226,88 @@ def test16_upsert_to_table_dummy(self):
...
+            schema=upsertTableSchema,
+            columns=["testid", "testdata"],
+            data=[(2, "newtestdata2"), (3, "testdata3")],
+        )
+        self.sql_server.upsert_to_table_by_name(
+            df_source=df_upsertTest,
+            table_name=self.table_upsert_name,
+            join_cols=["testid"],
+            filter_join_cols=False,
+            overwrite_if_target_is_empty=False,
```
Comment (LauJohansson) [inline tests/cluster/sql/test_deliverysql.py:300]: The name is "test19_upsert_to_table_overwrite_empty_target" but the overwrite_if_target_is_empty=false. It this intended?

Comment (JeppeBlixen) [reply tests/cluster/sql/test_deliverysql.py]: Fixed. But it is actually very hard to fix the overwrite scenario. The end state is the same as a merge, and the temp table is cleaned.

**Comment by LauJohansson** [comment]
All in all, I think it is great improvements. It alligns a lot with the delta upsert. I did not have any specific change requirements.

File: tests/cluster/sql/test_deliverysql.py (hunk 1 above)
Comment (LauJohansson) [reply tests/cluster/sql/test_deliverysql.py]: I agree.

**Review by LauJohansson — APPROVED** [review]

**Pull request merged**